hashintel/hash
# Fix table block inside the Hash editor

## 🌟 What is the purpose of this PR?




Previous PRs like #188 modified the table block and related `linkAggregation` queries in order to make the table block work with the new links system in Hash. However, the table block wasn't completely usable, because some essential functionality was missing in the Hash editor itself, and some unresolved GraphQL errors were present on the table block itself.

This PR takes care of both, to make the table block completely functional inside of the Hash Editor.

Changes are covered in GH comments below.

## 🛡 What tests cover this?



- This PR primarily depends on Playwright tests, TS type-checks, and Eslint linting for automated testing.

## ❓ How to test this?


> Pre-requisites: Serve the updated table block locally, and point to it from `blockPaths.json` to avoid any unexpected behavior while testing.

1. Checkout the branch locally.
2. Run `yarn` on the root of the repo.
3. Build and serve the new table block. Point to it from `blockPaths.json`.
4. Go to the Hash Editor.
5. Create a new table block, and try changing Entity types.
6. Check for any errors. If you face any, comment them here.

## Demo
![chrome_PxKbXhD5Nc](https://user-images.githubusercontent.com/22196279/159914858-1449817b-7102-49e1-a670-3bcbbd982e2e.gif)



---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201921476656467
